### DISCUSSION TITLE
Newsletter connection plugin SMTP or Webhook

### DISCUSSION THREAD

**Discussion by treeandcoffee** [GetPublii/Publii#1836] [Ideas]
I suggest a newsletter connection plugin which let you define newsletter settings and connectio for either custom SMTP or better for Webhooks. So you could define connection for forms for mailchimp, mailerlite or emailit directly.

**Comment by candidexmedia** [comment]
How do you imagine webhooks or a custom SMTP working in this scenario? Would it send an email every time a post is published?

If so, I'd suggest using your site's RSS feed as an automation trigger instead. It's how a service like [Mastofeed](https://mastofeed.org/) can automatically make a Mastodon "toot" every time you publish a new post / add an entry to your RSS, leveraging [POSSE](https://indieweb.org/POSSE).

For newsletters specifically: services like [buttondown](https://docs.buttondown.com/rss-to-email) and Mailchimp have RSS-to-email built-in, making it a ready made integration with your Publii site (and I'm sure there are [other services](https://fliprss.com/rss-to-email/) that can take care of that, since RSS is pretty widespread).

**Comment by treeandcoffee** [comment]
The idea went in a completely different direction and is only aimed at newsletters, namely as a plugin where you store your newsletter configuration. So that a newsletter form could be sent. The config would then not only be possible as an SMTP config, but also as an API definition of a webhook, so that make or Zapier or others could be used.

I didn't even think about the benefit you mentioned, that posts can be sent by email. But I agree with you that RSS and/or toots would be the right format here.
